controllers/datamover.go

```diff
@@ -792,3 +803,78 @@ func (r *DPAReconciler) parseGCPSecret(secret corev1.Secret, secretKey string) (
 
 	return gcpcreds, nil
 }
+
+func (r *DPAReconciler) ReconcileDataMoverResticCustomCA(log logr.Logger) (bool, error) {
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	backupStorageLocationList, err := r.listBackupStorageLocations()
+	if err != nil {
+		return false, err
+	}
+
+	if r.checkIfDataMoverIsEnabled(&dpa) {
+		for _, bsl := range backupStorageLocationList.Items {
+			if strings.Contains(bsl.Name, dpa.Name) {
```

Comment:
Please add a TODO here to change the logic when we allow users to specify BSL name

Reply:
also it might be more efficient to check the other way around.. dpa.spec.backupLocations instead of listing all BSLs that may contain those created outside of DPA.